base/applications/rapps/appview.cpp

```diff
@@ -1892,6 +1892,38 @@ CApplicationView::AppendTabOrderWindow(int Direction, ATL::CSimpleArray<HWND> &T
     m_AppsInfo->AppendTabOrderWindow(Direction, TabOrderList);
 }
 
+VOID
+CApplicationView::GetRestoreListSelectionData(LVITEMW &Item, WCHAR *Name, UINT NameLen)
+{
+    Item.mask = LVIF_TEXT|LVIF_STATE;
+    Item.iItem = -1, Item.iSubItem = 0;
+    Item.stateMask = LVIS_FOCUSED|LVIS_SELECTED;
+    Item.pszText = Name, Item.cchTextMax = NameLen;
+
+    HWND hList = m_ListView ? m_ListView->m_hWnd : NULL;
+    if (hList)
+    {
+        Item.iItem = ListView_GetNextItem(hList, -1, LVNI_FOCUSED);
+        ListView_GetItem(hList, &Item);
+    }
+}
+
+int
+CApplicationView::RestoreListSelection(const LVITEMW &Item)
+{
+    LVFINDINFOW fi;
+    fi.flags = LVFI_STRING;
+    fi.psz = Item.pszText;
+
+    HWND hList = m_ListView ? m_ListView->m_hWnd : NULL;
+    int index = hList ? ListView_FindItem(hList, -1, &fi) : -1;
+    if (index != -1)
+    {
+        ListView_SetItemState(hList, index, Item.state, Item.stateMask);
+    }
+    return index;
```

Comment:
Unless I missed something in the review, it doesn't seem that this returned index is used anywhere in the code. Is it still wanted to return the index "in case it could be useful in the future"? If so, what scenarii could make use of it?

Reply:
No it is not used now but it is useful to implement the feature where the uninstaller fails to remove its entry and the UI asks you if you want to remove the dead registry entry. That was my plan anyway.

Reply:
I removed it, can add it back in future PR.